pkg/backup/restore_data_processor.go

```diff
@@ -191,6 +191,11 @@ func (rd *restoreDataProcessor) Start(ctx context.Context) {
 
 	ctx, cancel := context.WithCancel(ctx)
 	rd.cancelWorkersAndWait = func() {
+		defer func() {
+			if p := recover(); p != nil {
```

Comment:
nit: can you add a TODO with my name on it to make it so panics in the restore processor do not cause the system to crash.

Reply:
Oh, I kind of thought a panic here would bring down the system anyway. I'll convert the `Fatalf` to rethrowing the panic with more detail.

Reply:
It will bring down the system, but I would like to normalize panic recovery, so I mostly want a reminder to myself to fix this in the future.

Reply:
Ok. Changed it to `panic()` and also added a TODO.

Reply:
Thanks!